src/Graphql.Extensions.FieldEnums/Types/FieldAwareGraphType.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Linq.Expressions;
+using GraphQL;
+using GraphQL.Builders;
+using Graphql.Extensions.FieldEnums.Types.Extensions;
+using GraphQL.Types;
+
+namespace Graphql.Extensions.FieldEnums.Types
+{
+    public class FieldAwareGraphType<TSourceType> : ObjectGraphType<TSourceType>
+    {
+        public override FieldBuilder<TSourceType, TProperty> Field<TProperty>
+        (
+            string name,
+            Expression<Func<TSourceType, TProperty>> expression,
+            bool nullable = false,
+            Type type = null
+        )
+        {
+            var result = base
+                .Field(name, expression, nullable, type)
+                .WithOriginalName(expression.NameOf())
```

Comment:
See graphql-dotnet/graphql-dotnet#1107, it will be better to use already implemented features.

Reply:
How do I have to use it? It wasn't "already implemented" when i initially did this last Winter.

Reply:
Even after #38 ?

Reply:
`ORIGINAL_EXPRESSION_PROPERTY_NAME` now internal, OK, I'll see what I can do.

Reply:
Hmm okay didn't notice the update, sorry my fault :D
But my solution still needs the sourcetype

Reply:
Now using the internal key (https://github.com/JKamsker/examples/blob/master/src/Graphql.Extensions.FieldEnums/SharedConstants.cs)